### DISCUSSION TITLE
Ho preso un miniPC, consigli configurazione

### DISCUSSION THREAD

**Discussion by chip83** [indomus/forum#2942] [HUB: Home Assistant]
Mi sono buttato anche io nel mondo della domotica. Ho acquistato un miniPC [consigliato sul sito](https://indomus.it/componenti/mini-pc-alder-lake-n95-agemagician-ace-ad03-8gb-ram-e-256gb-ssd/), con N95+8GB di ram.
Vorrei farci un po' di cose...
la cosa primaria sarà **Home Assistant** per gestione di luci (shelly), sensori di temperatura ZigBee SONOFF, Dongle SONOFF ZigBee, 2 telecamere EZVIZ, impianto fotovoltaico con batterie SENEC Smathome V3 hybrid, Pompe di calore con Broadlink Smart Home Hub, RM Mini3). I sensori ZigBee dovranno quindi funzionare tramite ZigBee2MQTT.

A questo vorrei aggiungere la riproduzione di contenuti multimediali presenti su HDD interno e/o penne USB/HDD e lo streaming di cotenuti video (1080p su TV). Inoltre vorrei anche fare retrogaming con Batocera ed un controller (che acquisterò successivamente) bluetooh.

Come mi consigliate di configurare il miniPC? HA core?
Come faccio? Non voglio dover riconfigurare tutto tra 2 mesi se magari voglio aggiungere Batocera o similari...

**Comment by indomus** [comment]
Parti da qui: https://indomus.it/formazione/home-assistant-su-mini-pc-intel-nuc-le-varie-installazioni-le-cose-da-fare/
Leggi TUTTO, ti si chiariranno molti concetti. Non c'è una risposta univoca, la strada la devi approcciare e capire da te.

**Reply by chip83** [reply]
si, quella guida l'avevo già letta (2 volte! :D).
Sostanzialmente la mia idea era installare Home Assistant Core su Docker.
Quello che non capisco è se, facendo così, tramite Docker, posso poi installare ZigBee2MQTT, Batocera e l'applicativo che mi servirà per lo streaming video.

**Reply by indomus** [reply]
batocera sicuramente no (ha bisogno di una sua partizione ad hoc, puoi usare anche una chiavetta interna o un secondo disco), tutto il resto tipicamente sì (incluso zigbee2mqtt)

**Reply by chip83** [reply]
quindi per Batocera dovrei installare un secondo HDD sata (oppure una chiavetta USB) ed avviarlo da li?
Questo ovviamente comporta che HA non mi funzionerà mentre utilizzo Batocera no?
Avevo letto da qualche parte che Batocera poteva funzionare anche su Docker, peccato :\

Ah, inoltre, per lo streaming video cosa mi consigli?

**Reply by indomus** [reply]
non ci risulta disponibile una versione docker di batocera, verifica ma dovrebbe essere così.
che intendi di preciso per streaming video?

**Reply by chip83** [reply]
vorrei visualizzare IPTV rumene/italiane

**Reply by indomus** [reply]
è off topic rispetto ai temi che seguiamo, comunque qualche applicativo certamente ci sarà. cerca su https://hub.docker.com/

**Comment by fbarachino** [comment]
Ciao, se hai un po di dimestichezza, potrei consigliarti di installarci Proxmox, e quindi creare una vm con homeassistant. Sia perchè puoi sfruttare il minipc per altre macchine virtuali, sia per la gestione di backup e ripristino delle stesse. Poi, come il consiglio di cui sopra, dipende dalla strada che vuoi scegliere...

Ciao
Flavio.

**Reply by indomus** [reply]
proxmox lo [sconsigliamo totalmente](https://indomus.it/focus/home-assistant-hassio-o-home-assistant-core-quale-installazione-fa-al-caso-mio/) 👍

**Reply by fbarachino** [reply]
Confermo che tecnicamente è un mappazzone di un container (docker) su vm (proxmox), ma se vuoi utilizzare il minipc anche per altro, sei costretto a tutte macchine docker.
Io comunque, nonostante tutto, ho un chuwi hero (visto la recensione su indomus.it) con su Proxmox e oltre alla VM di homeassistant os, ho anche altre 10 macchine tra cui anche pfSense che mi fa da router per la fibra. E devo dire che nonostante tutto gira tutto molto bene e veloce.

Poi, ripeto, questa è una mia oppinione. Sicuramente non è vangelo. 
E sicuramente è meglio ascoltare inDomus che ha sicuramente più esperienza di me in questo campo. 

Ciao

**Reply by indomus** [reply]
perdonami: chi ti vieta di usare contemporaneamente docker (per esempio per Home Assistant Core) e proxmox per altro che davvero abbia bisogno di una macchina virtuale piuttosto che di uno snello container? proxmox non è il male, anzi, però può convivere con altro (e va usato secondo noi quando davvero ha senso farlo).

**Reply by fbarachino** [reply]
Ciao, 
Scusami, ma voglio innanzitutto chiarire che non vorrei accendere una discussione su questo argomento . La mia era solo un punto di vista e nient'altro. 
Sicuramente è possibile fare come dici, installo docker e proxmox sulla stessa macchina fisica (anche se sinceramente, a me personalmente, non piace) ma appunto le vie sono infinite. 
A me, sinceramente, Docker non piace, ma è una questione di gusti la mia. 
E alla fine condivido a pieno l'articolo che mi hai linkato prima (anche se a suo tempo ho disubbidito... ;) )

Spero che si sia capito che per me era solo una questione di gusti e (a mio parere ) di praticità (e non prettamente tecnica).

**Reply by indomus** [reply]
no ma anzi è un interessante argomento da dibattere, proprio perché molti utenti si trovano a passare di qui e magari possiamo essere di spunto per qualcuno a venire ;)

**Reply by snakuzzo** [reply]
> Confermo che tecnicamente è un mappazzone di un container (docker) su vm (proxmox), ma se vuoi utilizzare il minipc anche per altro, sei costretto a tutte macchine docker

Non esistono macchine docker, semmai container Docker...che non sono macchine (o VM)

**Reply by trebestie** [reply]
Io ho seguito la cattiva strada.
Container lxc/incus, al momento ne ho 13 attivi, e homeassistant core su una macchina ubuntu, Z2m e mqtt dockerizzati ognuno nel proprio container.
Motivi?
Gestione della rete più ordinata, massima flessibilità e libertà nell'utilizzo delle applicazioni,  uso docker se mi serve altrimenti vado liscio, e dei sistemi operativi, posso fare tutti i pasticci che voglio senza compromettere le componenti importanti, snapshot e backup per ripristinare in caso di guai.
Aspetti negativi?
Il sistema è abbastanza complesso e non è semplice da tenere aggiornato.
Risorse hardware?
Ho un Fujitsu Futro 720 con 4MB comprato usato su ebay cui ho aggiunto un ssd da 500GB 
Problemi riscontrati?
Non sempre al riavvio i device usb prendono lo stesso numero, al momento non ho ancora trovato come risolvere definitivamente il problema e ne posso usare uno solo

**Comment by fbarachino** [comment]
Ops.. lapsus... chiedo venia.
